mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/dirtiness/building/impl/PojoIndexingDependencyCollectorValueNode.java

```diff
@@ -59,18 +60,29 @@
 	private final PojoIndexingDependencyCollectorTypeNode<?> lastEntityNode;
 	private final BoundPojoModelPathValueNode<?, P, V> modelPathFromLastEntityNode;
 
+	private final ReindexOnUpdate reindexOnUpdate;
+
 	// First key: inverse side entity type, second key: original side concrete entity type
 	private Map<PojoRawTypeModel<?>, Map<PojoRawTypeModel<?>, PojoModelPathValueNode>> inverseAssociationPathCache =
 			new HashMap<>();
 
-	PojoIndexingDependencyCollectorValueNode(BoundPojoModelPathValueNode<?, P, V> modelPathFromRootEntityNode,
+	PojoIndexingDependencyCollectorValueNode(PojoIndexingDependencyCollectorPropertyNode<?, P> parentNode,
+			BoundPojoModelPathValueNode<?, P, V> modelPathFromRootEntityNode,
 			PojoIndexingDependencyCollectorTypeNode<?> lastEntityNode,
 			BoundPojoModelPathValueNode<?, P, V> modelPathFromLastEntityNode,
 			PojoImplicitReindexingResolverBuildingHelper buildingHelper) {
 		super( buildingHelper );
 		this.modelPathFromRootEntityNode = modelPathFromRootEntityNode;
 		this.lastEntityNode = lastEntityNode;
 		this.modelPathFromLastEntityNode = modelPathFromLastEntityNode;
+
+		BoundPojoModelPathPropertyNode<?, P> modelPathPropertyNode = modelPathFromRootEntityNode.getParent();
+		this.reindexOnUpdate = buildingHelper.getReindexOnUpdate(
+				parentNode.getReindexOnUpdate(),
+				modelPathPropertyNode.getParent().getTypeModel(),
```

Comment:
I'm surprised you use `getParent()` here knowing you already called it once above?

Reply:
I'm accessing the grandparent here. I'll change the code to make it clearer.

Reply:
I confirm it's better now.